### PR TITLE
Migrate to Kotlin

### PR DESCRIPTION
**Pull request by marc0der**
This achieves the following:
* migrate the `AppRepo` class to Kotlin
* migrate `MongoHealthCheck` to Kotlin
* Remove obsolete `MongoHealthCheck` unit test (already covered by an acceptance test)

### CODE REVIEW

File: src/main/kotlin/io/sdkman/broker/app/AppRepo.kt [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+import ratpack.exec.Blocking
+import ratpack.exec.Promise
+import javax.inject.Inject
+import javax.inject.Singleton
+
+@Singleton
+open class AppRepo @Inject constructor(private val mongoProvider: MongoProvider) {
+
+    private val applicationCollectionName = "application"
+    private val aliveFieldName = "alive"
```
Comment (helpermethod) [inline src/main/kotlin/io/sdkman/broker/app/AppRepo.kt]: Should probably be a top-level `const`.

File: src/main/kotlin/io/sdkman/broker/app/AppRepo.kt [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+import io.sdkman.broker.db.MongoProvider
+import ratpack.exec.Blocking
+import ratpack.exec.Promise
+import javax.inject.Inject
+import javax.inject.Singleton
+
+@Singleton
+open class AppRepo @Inject constructor(private val mongoProvider: MongoProvider) {
+
+    private val applicationCollectionName = "application"
```
Comment (helpermethod) [inline src/main/kotlin/io/sdkman/broker/app/AppRepo.kt]: Should probably be a top-level `const`.

File: src/main/kotlin/io/sdkman/broker/health/MongoHealthCheck.kt [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+import ratpack.health.HealthCheck
+import ratpack.health.HealthCheck.Result
+import ratpack.registry.Registry
+import javax.inject.Inject
+import javax.inject.Singleton
+
+@Singleton
+class MongoHealthCheck @Inject constructor(private val appRepo: AppRepo) : HealthCheck {
+
+    private val unhealthyMessage = "Nothing found at application/alive in database."
```
Comment (helpermethod) [inline src/main/kotlin/io/sdkman/broker/health/MongoHealthCheck.kt]: Same as above. I think if declared const `unhealthyMessage` gets inlined by the compiler.

**Review by helpermethod — COMMENTED** [review]
Only a few minor quibbles.

File: src/main/kotlin/io/sdkman/broker/health/MongoHealthCheck.kt (hunk 3 above)
Comment (marc0der) [reply src/main/kotlin/io/sdkman/broker/health/MongoHealthCheck.kt]: I don't feel so strongly about this, as in all these cases the entire class is a `@Singleton`, but I've changed all these fields to `const` now.

**Review by helpermethod — APPROVED** [review]
LGTM!

**Pull request merged**